lib/internal/test_runner/test.js

```diff
@@ -263,6 +263,16 @@ class TestContext {
           return ReflectApply(method, this, args);
         };
       });
+
+      // assert.ok needs it's own function, as it checks the stack
+      // trace to provide more descriptive errors.
+      function ok(...args) {
```

Comment:
This should go in the `lazyAssertObject()` function and not duplicate the logic for tracking the plan.

Reply:
assert.ok needs it's own function, as it uses itself as the reference point for tracking the stacktrace. If it went in that function, it would have another wrapper around it, breaking that?

Reply:
It shouldn't? Doesn't "needs it's own function" just mean that `ok` is passed as the first argument to `innerOk`?

Reply:
Yes, `ok` is passed as the first argument, but if it's created in the `lazyAssertObject` function, the `get assert()` will make a new wrapper around it for test planning, meaning the function passed to `innerOk` isn't the same as the function that is in use.

Reply:
That's frustrating. Please leave a comment that this is a hack and hopefully it can be revisited in the future.

Reply:
I've added a TODO comment for this logic to be implemented via better handling in the `innerOk` function. How does it look?